hawkauthlib/tests/test_signatures.py

```diff
@@ -63,13 +63,69 @@ def test_get_signature_by_parsing_authz_header(self):
         algorithm = "sha256"
         sig = "6R4rV5iE+NPoym+WwjeHzjAGXUtLNIxmo1vpMofpLAE="
         mysig = get_signature(req, key, algorithm)
-        self.assertEquals(sig, mysig)
+        self.assertEqual(sig, mysig)
+
+    def test_post_signature_by_parsing_authz_header(self):
+        req = b"POST /resource/1?b=1&a=2 HTTP/1.1\r\n"\
+              b"Host: example.com:8000\r\n"\
+              b"Content-Type: application/json; charset=utf-8\r\n"\
+              b"Content-Length: 26\r\n"\
+              b"Authorization: Hawk "\
+              b"    id=\"dh37fgj492je\","\
+              b"    ts=\"1353832234\","\
+              b"    nonce=\"j4h3g2\","\
+              b"    ext=\"some-app-ext-data\""\
+              b"\r\n\r\n{\"test\": \"just some text\"}"
+        key = "werxhqb98rpaxn39848xrunpaw3489ruxnpa98w4rxn"
+        algorithm = "sha256"
+        sig = "56wgBMHr4oIwA/dGZspMm6Zk4rnf3aiwwVeL0VtWoGo="
+        mac = get_signature(req, key, algorithm)
+        self.assertEqual(sig, mac)
+
+    def test_post_with_payload_verification(self):
+        req = b"POST /resource/1?b=1&a=2 HTTP/1.1\r\n"\
+              b"Host: example.com:8000\r\n"\
+              b"Content-Type: application/json; charset=utf-8\r\n"\
+              b"Content-Length: 26\r\n"\
+              b"Authorization: Hawk "\
+              b"    id=\"dh37fgj492je\","\
+              b"    ts=\"1353832234\","\
+              b"    nonce=\"j4h3g2\","\
+              b"    hash=\"Lj826IMSgrm1vnCFQNxaXasE8zHBXSWq2pPtnHEmaO8=\","\
+              b"    ext=\"some-app-ext-data\""\
+              b"\r\n\r\n{\"test\": \"just some text\"}"
+        key = "werxhqb98rpaxn39848xrunpaw3489ruxnpa98w4rxn"
+        algorithm = "sha256"
+        sig = "Tx7PoLWYtn3VJEc0GdkguHkEB281grYEtFycPhtln9w="
+        mac = get_signature(req, key, algorithm)
+        self.assertEqual(sig, mac)
+
+    def test_version_2_0_0_post_signature_by_parsing_authz_header(self):
```

Comment:
@rfk I added a test to verify backwards compatibility of `get_signature` to version 2.0.0